File: config/initializers/stripe.rb
Rails.configuration.stripe = {
  publishable_key: ENV['pk_test_51NdRbxFktg0CjVwYfy15P3pYZanFGMinl4R7xtKZnaPvGskTIqYXKZTR1KnuNgv7e6DNDzxHDJSsq8VMejK9u6Jn00rx5EcpFO'],
  secret_key: ENV['sk_test_51NdRbxFktg0CjVwY3B3Ivlf0w3RmJibCuuXuRJG8IxuUHhJD1ywExOqIRjBgmtjdTaXJ262XHRGfs3jWWUH0T04E00nFkSeCZy']
}

Stripe.api_key = Rails.configuration.stripe[:secret_key]

File: app/assets/javascripts/stripe.js
document.addEventListener("DOMContentLoaded", function () {
  var stripe = Stripe("<%= Rails.configuration.stripe[:publishable_key] %>");
  var elements = stripe.elements();

  var card = elements.create("card");
  card.mount("#card-element");

  var cardErrors = document.getElementById("card-errors");

  card.on("change", function (event) {
    if (event.error) {
      cardErrors.textContent = event.error.message;
    } else {
      cardErrors.textContent = "";
    }
  });

  var form = document.getElementById("payment-form");
  form.addEventListener("submit", function (event) {
    event.preventDefault();

    stripe.createToken(card).then(function (result) {
      if (result.error) {
        cardErrors.textContent = result.error.message;
      } else {
        var token = result.token.id;

        $.post("/checkout/process_payment", { token: token }, function(data) {


        });
      }
    });
  });
});
